core/src/executor/select/mod.rs

```diff
@@ -131,6 +131,44 @@ pub async fn select_with_labels<'a>(
                 .into_iter()
                 .map(|i| format!("column{}", i))
                 .collect::<Vec<_>>();
+
+            let mut column_types = values_list[0]
+                .iter()
+                .map(|expr| {
+                    evaluate_stateless(None, expr)
+                        .and_then(|evaluated| evaluated.try_into())
+                        .map(|value: Value| value.get_type())
+                })
+                .collect::<Result<Vec<_>>>()?;
+
+            let get_null_indexes = |types: &Vec<Option<DataType>>| -> Vec<usize> {
+                types
+                    .iter()
+                    .enumerate()
+                    .filter(|(_, t)| t.is_none())
+                    .map(|(i, _)| i)
+                    .collect::<Vec<_>>()
+            };
+
+            for exprs in values_list {
+                let null_indexes = get_null_indexes(&column_types);
+                if null_indexes.is_empty() {
+                    break;
+                };
+                exprs
+                    .iter()
+                    .enumerate()
+                    .filter(|(i, _)| null_indexes.contains(i))
+                    .try_for_each(|(i, expr)| -> Result<()> {
+                        let value: Value = evaluate_stateless(None, expr)?.try_into()?;
+                        if let Some(data_type) = value.get_type() {
+                            column_types[i] = Some(data_type);
+                        }
+
+                        Ok(())
+                    })?;
+            }
+
```

Comment:
```rust
            let rows = values_list.into_iter().scan(
                iter::repeat(None)
                    .take(first_len)
                    .collect::<Vec<Option<DataType>>>(),
                move |column_types, exprs| {
                    if exprs.len() != first_len {
                        return Some(Err(RowError::NumberOfValuesDifferent.into()));
                    }

                    let values = column_types
                        .iter_mut()
                        .zip(exprs.iter())
                        .map(|(column_type, expr)| -> Result<_> {
                            let evaluated = evaluate_stateless(None, expr)?;

                            let value = match column_type {
                                Some(data_type) => evaluated.try_into_value(data_type, true)?,
                                None => {
                                    let value: Value = evaluated.try_into()?;
                                    *column_type = value.get_type();

                                    value
                                }
                            };

                            Ok(value)
                        })
                        .collect::<Result<Vec<_>>>()
                        .map(Row);

                    Some(values)
                },
            );
```

Since it runs on all `values_list` below, I think it can be integrated.
This eliminates pseudo `unwrap` syntax, such as `[index]` access, and eliminates duplicate execution.

How about this code?